src/sagas.test.js

```diff
@@ -0,0 +1,263 @@
+import {all, call, put, select, takeEvery} from 'redux-saga/effects';
+import {cloneableGenerator} from 'redux-saga/utils';
+import {
+  checkStatus,
+  getOngoingVersions,
+  getPollbotVersion,
+  getReleaseInfo,
+} from './PollbotAPI';
+import {
+  addCheckResult,
+  setVersion,
+  updateLatestChannelVersions,
+  updatePollbotVersion,
+  updateReleaseInfo,
+} from './actions';
+import {
+  REFRESH_STATUS,
+  REQUEST_ONGOING_VERSIONS,
+  REQUEST_POLLBOT_VERSION,
+  REQUEST_STATUS,
+  UPDATE_URL,
+} from './types';
+import {
+  fetchOngoingVersions,
+  fetchPollbotVersion,
+  refreshStatus,
+  requestStatus,
+  rootSaga,
+  updateUrl,
+} from './sagas';
+
+describe('sagas', () => {
+  it('handles fetchPollbotVersion', () => {
+    const data = {};
+    data.saga = cloneableGenerator(fetchPollbotVersion)();
+
+    const pollbotVersion = {
+      name: 'pollbot',
+      source: 'https://github.com/mozilla/PollBot.git',
+      version: '0.2.1-22-g8e09a0f',
+      commit: '8e09a0f8e995344ea24fbb940a6bddc17e0edaed',
+    };
+    expect(data.saga.next().value).toEqual(call(getPollbotVersion));
+
+    // Clone to test success and failure of getPollbotVersion.
+    data.sagaThrow = data.saga.clone();
+
+    expect(data.saga.next(pollbotVersion).value).toEqual(
+      put(updatePollbotVersion(pollbotVersion)),
+    );
+    expect(data.saga.next().done).toBe(true);
+
+    expect(data.sagaThrow.throw('error').value).toEqual(
+      console.error('Failed getting the pollbot version', 'error'),
```

Comment:
How does this work? Won't this just evaluate the call to `console.error` and return undefined (or whatever `console.error` returns)?

Reply:
I totally overlooked that (it returns `undefined`). I'll modify to instead mock the `console.error` function